Lib/ensurepip/__init__.py

```diff
@@ -26,9 +27,17 @@ def _run_pip(args, additional_paths=None):
     if additional_paths is not None:
         sys.path = additional_paths + sys.path
 
-    # Install the bundled software
-    import pip._internal
-    return pip._internal.main(args)
+    # Invoke pip as if it's the main module, and catch the exit.
+    backup_argv = sys.argv[:]
+    sys.argv[1:] = args
+    try:
+        runpy.run_module("pip", run_name="__main__", alter_sys=True)
+    except SystemExit as e:
+        return e.code
+    finally:
+        sys.argv[:] = backup_argv
+
+    raise SystemError("pip have not exited, that should never happen")
```

Comment:
Honestly, I don't think that it's a big deal that pip `__main__` doesn't raise SystemExit in some cases. I suggest to just use `return 0` in this case.
```suggestion
    # run_module() didn't raise SystemExit: success
    return 0
```

Reply:
This was requested by @ncoghlan

Reply:
@ncoghlan: What is the point of raising an hard error if run_module()? For me, it's safe to consider that it's a success and just exit the process with exit code 0 (success).

cc @pradyunsg

Reply:
Honestly, I don't care what we do here.

pip has an unconditional `sys.exit()` call, so we're really not hit this branch, like, ever on any normal system/setup.

Reply:
@vstinner In that case, I really rather blow up than continue it there was no exit.